shared/aux.go

```diff
@@ -69,14 +70,12 @@ func RunCommandOnNode(cmd, ip string) (string, error) {
 	cleanedStderr := strings.ReplaceAll(stderr, "\n", "")
 	cleanedStderr = strings.ReplaceAll(cleanedStderr, "\t", "")
 
-	if cleanedStderr != "" && (!strings.Contains(stderr, "exited") ||
-		!strings.Contains(cleanedStderr, "1") ||
+	if cleanedStderr != "" && (!strings.Contains(stderr, "exited") || !strings.Contains(cleanedStderr, "1") ||
 		!strings.Contains(cleanedStderr, "2")) {
 		return cleanedStderr, nil
 	} else if cleanedStderr != "" {
 		return "", fmt.Errorf("command: %s failed with error: %v\n", cmd, stderr)
 	}
-	LogLevel("debug", fmt.Sprintf("StdOut: %s", stdout))
 
```

Comment:
if we dont want something logged, paritcularly if you feel it is senstive, adding a log:true/false descision maker can be helpful and leave to true by default for the method.

Reply:
i  disagree in 2 important aspects

1- https://go.dev/play/p/xff8eeuNwAL , run this and the results will tell by itself

2- who should decide which to log or not is the consumer always is this case of a return 
As a consumer that needs to invoke that RUN function
i want to be able to manage my return data
So that i can have control of my stuff

Reply:
now if you want more verbose WHEN u are developing something , please add prints and log everywhere, i do the same
but when its complete, you should log a lot, agree, but with cautious

Reply:
if we pass logging the stdout to the testcases, it will be adding a 100 line to every testcase change. every step u test the error the next step add a log the stdout. it makes the testcase noisy. Not only the testcase, inside the core shared methods as well. 
For example, the secrets encryption status output or the manage service... etc etc. every place, we ll have to go back and add an stdout log. if you remove this change, that will literally have to be be my next PR. 
instead, if we keep this here, it is one place logging it all. 
This is a stickler for me about more logging. We seem to avoid any logging everywhere, to be honest, it is very hard to read through the logs. This line is a centralized one. 
Please feel free to add logStdOut:true or false decision maker passed into the method as an argument for sensitive stuff. But please do not remove this.

Reply:
it is not about development. when am developing i know whats going on. its hard to read the logs from jenkins jobs of other people code. and hence the need and necessity for more verbose logging.

Reply:
i searched the "RunCommand" in our code, right now, we use it in 73 places. so that will be literally adding 73 lines of LogLevel("DEBUG", "Logging stdout") in 73 places... instead of 1 log stdout.

Reply:
i believe you miss the point maybe, in every call you make to this func, the func already returns the output right ? 
Lets get back a little 

Regarding design is one thing

About adding a log is another, since this func HAS a return which is the stdout that u want to log.
if you run this - https://go.dev/play/p/xff8eeuNwAL

you will see the output 2 times, right ? thats because you are logging with no need, since the func itself already returns that stuff for you.

if you still dont understood lets sync in a call

Reply:
> if you remove this change, that will literally have to be be my next PR.


why do you need to print the output?

Reply:
i just got an example from your test , if you see here you  are also already logging the output!, thats the correct usage, why duplicate logs with no need ? 


```
func verifyTLSDirContent(product string, ips []string) {
	dataDir := fmt.Sprintf("/var/lib/rancher/%s", product)
	serverDir := fmt.Sprintf("%s/server", dataDir)
	origTLSDir := fmt.Sprintf("%s/tls", serverDir)

	cmd := fmt.Sprintf("sudo ls -lt %s/ | grep tls | awk {'print $9'} | sed -n '2 p'", serverDir)

	for _, ip := range ips {
		shared.LogLevel("info", fmt.Sprintf("Working on node with ip: %s", ip))

		tlsDir, tlsError := shared.RunCommandOnNode(cmd, ip)
		Expect(tlsError).NotTo(HaveOccurred(), fmt.Sprintf("Unable to get new TLS Directory name for %s", ip))

		shared.LogLevel("info", fmt.Sprintf("TLS Directory name: %s", tlsDir))
		newTLSDir := fmt.Sprintf("%s/%s", serverDir, tlsDir)

		shared.LogLevel("info", "Comparing Directories: %s and %s", origTLSDir, newTLSDir)
		cmd2 := fmt.Sprintf("sudo diff -sr %s/ %s/ | grep -i identical | "+
			"awk '{print $2}' | xargs basename -a | "+
			"awk 'BEGIN{print \"Identical Files:  \"}; {print $1}'", origTLSDir, newTLSDir)

		identicalFileList, err := shared.RunCommandOnNode(cmd2, ip)
		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("error getting identical files on %s", ip))

		shared.LogLevel("debug", fmt.Sprintf("Identical Files Output for %s: %s", ip, identicalFileList))

		verifyIdenticalFiles(identicalFileList)
	}
```

Reply:
You are right. logging them where we use it with more context does make it better readable :) 👍

Reply:
This is a really good discussion, and made me think of the larger issue here in that having some logging at different levels, specifically debug, could be useful in critical places if/when tests need to get reworked. The problem is that we just set the loglevel to be debug always, so logs get overwhelming and up just being like print statements everywhere, which gets too verbose and is unnecessary. 

I created https://github.com/rancher/qa-tasks/issues/1232 to resolve that so that we can move forward with the best of both worlds -- having some debug logging in critical places and info logging where we want prints generally.